client/push/push_manager.js

```diff
@@ -25,9 +25,36 @@ export default class Push {
   }
 
   /**
+   *  get push permission status
+   */
+  getPushPermissionStatus () {
+    // query push permissions
+    return new Promise(function(resolve, reject) {
+      navigator.permissions.query({name:'push', userVisibleOnly:true})
+      .then(permissionStatus => {
+        console.log('push permission state is ', permissionStatus.state);
+        resolve(permissionStatus.state);
+      });
+    });
+  }
+
+  /**
+   * get push subscription from push manager
+   */
+  getPushSubscription() {
```

Comment:
Where is this function used?

Reply:
I added this function to check push was subscribed or not before calling registerServiceWorker function.
not yet implemented.